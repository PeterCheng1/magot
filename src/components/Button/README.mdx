---
name: Button 按钮
route: /components/Button
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Button from './Button';

# Button 按钮

按钮组件，用于响应用户的点击交互行为，触发特定的业务逻辑。

## Usage

### 按钮类型

<Playground>
  <Button>Default</Button>
  <Button type="primary">Primary</Button>
  <Button type="success">Success</Button>
  <Button type="warning">Warning</Button>
  <Button type="danger">Danger</Button>
  <Button type="text">Text</Button>
  <Button type="link">Link</Button>
</Playground>

### 按钮尺寸

<Playground>
  <div className="demo-group">
    <Button size="small">Small</Button>
    <Button>Normal</Button>
    <Button size="large">Large</Button>
  </div>
  <div className="demo-group">
    <Button type="primary" size="small">
      Small
    </Button>
    <Button type="primary">Normal</Button>
    <Button type="primary" size="large">
      Large
    </Button>
  </div>
  <div className="demo-group">
    <Button type="success" size="small">
      Small
    </Button>
    <Button type="success">Normal</Button>
    <Button type="success" size="large">
      Large
    </Button>
  </div>
  <div className="demo-group">
    <Button type="warning" size="small">
      Small
    </Button>
    <Button type="warning">Normal</Button>
    <Button type="warning" size="large">
      Large
    </Button>
  </div>
  <div className="demo-group">
    <Button type="danger" size="small">
      Small
    </Button>
    <Button type="danger">Normal</Button>
    <Button type="danger" size="large">
      Large
    </Button>
  </div>
  <div className="demo-group">
    <Button type="text" size="small">
      Small
    </Button>
    <Button type="text">Normal</Button>
    <Button type="text" size="large">
      Large
    </Button>
  </div>
  <div className="demo-group">
    <Button type="link" size="small">
      Small
    </Button>
    <Button type="link">Normal</Button>
    <Button type="link" size="large">
      Large
    </Button>
  </div>
</Playground>

### 不可用状态

<Playground>
  <div className="demo-group">
    <Button>Default</Button>
    <Button disabled={true}>Default(disabled)</Button>
  </div>
  <div className="demo-group">
    <Button type="primary">Primary</Button>
    <Button type="primary" disabled={true}>
      Primary(disabled)
    </Button>
  </div>
  <div className="demo-group">
    <Button type="success">Success</Button>
    <Button type="success" disabled={true}>
      Success(disabled)
    </Button>
  </div>
  <div className="demo-group">
    <Button type="warning">Warning</Button>
    <Button type="warning" disabled={true}>
      Warning(disabled)
    </Button>
  </div>
  <div className="demo-group">
    <Button type="danger">Danger</Button>
    <Button type="danger" disabled={true}>
      Danger(disabled)
    </Button>
  </div>
  <div className="demo-group">
    <Button type="text">Text</Button>
    <Button type="text" disabled={true}>
      Text(disabled)
    </Button>
  </div>
  <div className="demo-group">
    <Button type="link" href="https://github.com/billjs/magot" target="_blank">
      Link
    </Button>
    <Button
      type="link"
      href="https://github.com/billjs/magot"
      target="_blank"
      disabled={true}>
      Link(disabled)
    </Button>
  </div>
</Playground>

### 图标按钮

<Playground>
  <div className="demo-group">
    <Button size="small" icon="download">
      Default
    </Button>
    <Button type="primary" size="small" icon="download">
      Primary
    </Button>
    <Button type="success" size="small" icon="smile">
      Success
    </Button>
    <Button type="warning" size="small" icon="return">
      Warning
    </Button>
    <Button type="danger" size="small" icon="delete">
      Danger
    </Button>
    <Button type="text" size="small" icon="comment">
      Text
    </Button>
    <Button type="link" size="small" icon="message">
      Link
    </Button>
  </div>
  <div className="demo-group">
    <Button icon="download">Default</Button>
    <Button type="primary" icon="download">
      Primary
    </Button>
    <Button type="success" icon="smile">
      Success
    </Button>
    <Button type="warning" icon="return">
      Warning
    </Button>
    <Button type="danger" icon="delete">
      Danger
    </Button>
    <Button type="text" icon="comment">
      Text
    </Button>
    <Button type="link" icon="message">
      Link
    </Button>
  </div>
  <div className="demo-group">
    <Button icon="download" size="large">
      Default
    </Button>
    <Button type="primary" size="large" icon="download">
      Primary
    </Button>
    <Button type="success" size="large" icon="smile">
      Success
    </Button>
    <Button type="warning" size="large" icon="return">
      Warning
    </Button>
    <Button type="danger" size="large" icon="delete">
      Danger
    </Button>
    <Button type="text" size="large" icon="comment">
      Text
    </Button>
    <Button type="link" size="large" icon="message">
      Link
    </Button>
  </div>
  <div className="demo-group">
    <Button disabled={true} icon="download">
      Default
    </Button>
    <Button type="primary" disabled={true} icon="download">
      Primary
    </Button>
    <Button type="success" disabled={true} icon="smile">
      Success
    </Button>
    <Button type="warning" disabled={true} icon="return">
      Warning
    </Button>
    <Button type="danger" disabled={true} icon="delete">
      Danger
    </Button>
    <Button type="text" disabled={true} icon="comment">
      Text
    </Button>
    <Button type="link" disabled={true} icon="message">
      Link
    </Button>
  </div>
</Playground>

#### 图标在右侧

<Playground>
  <Button iconPosition="right" icon="download">
    Default
  </Button>
  <Button type="primary" iconPosition="right" icon="download">
    Primary
  </Button>
  <Button type="success" iconPosition="right" icon="smile">
    Success
  </Button>
  <Button type="warning" iconPosition="right" icon="return">
    Warning
  </Button>
  <Button type="danger" iconPosition="right" icon="delete">
    Danger
  </Button>
  <Button type="text" iconPosition="right" icon="comment">
    Text
  </Button>
  <Button type="link" iconPosition="right" icon="message">
    Link
  </Button>
</Playground>

#### 纯图标按钮

<Playground>
  <Button icon="download" />
  <Button type="primary" icon="download" />
  <Button type="success" icon="smile" />
  <Button type="warning" icon="return" />
  <Button type="danger" icon="delete" />
  <Button type="text" icon="comment" />
  <Button type="link" icon="message" />
</Playground>

#### 圆角纯图标按钮

<Playground>
  <Button icon="scan" circular={true} size="small" />
  <Button icon="scan" circular={true} />
  <Button icon="scan" circular={true} size="large" />
  <Button type="primary" icon="download" circular={true} size="small" />
  <Button type="primary" icon="download" circular={true} />
  <Button type="primary" icon="download" circular={true} size="large" />
  <Button type="danger" icon="delete" circular={true} size="small" />
  <Button type="danger" icon="delete" circular={true} />
  <Button type="danger" icon="delete" circular={true} size="large" />
</Playground>

#### 方形纯图标按钮

<Playground>
  <Button icon="scan" square={true} size="small" />
  <Button icon="scan" square={true} />
  <Button icon="scan" square={true} size="large" />
  <Button type="primary" icon="download" square={true} size="small" />
  <Button type="primary" icon="download" square={true} />
  <Button type="primary" icon="download" square={true} size="large" />
  <Button type="danger" icon="delete" square={true} size="small" />
  <Button type="danger" icon="delete" square={true} />
  <Button type="danger" icon="delete" square={true} size="large" />
</Playground>

#### 加载中按钮

<Playground>
  <div className="demo-group">
    <Button loading={true}>Loading</Button>
    <Button type="primary" loading={true}>
      Loading
    </Button>
    <Button type="text" loading={true}>
      Loading
    </Button>
    <Button loading={true} circular={true} />
    <Button type="primary" loading={true} circular={true} />
  </div>
</Playground>

### Block 按钮

<Playground>
  <Button block={true}>Default</Button>
  <Button type="primary" block={true}>
    Primary
  </Button>
  <Button type="success" block={true}>
    Success
  </Button>
  <Button type="warning" block={true}>
    Warning
  </Button>
  <Button type="danger" block={true}>
    Danger
  </Button>
  <Button type="text" block={true}>
    Text
  </Button>
  <Button type="link" block={true}>
    Link
  </Button>
</Playground>

### 按钮点击交互

<Playground>
  <Button onClick={() => alert('You clicked the Button')}>Click me</Button>
  <Button type="primary" onClick={() => alert('You clicked the Button')}>
    Click me
  </Button>
  <Button type="success" onClick={() => alert('You clicked the Button')}>
    Click me
  </Button>
  <Button type="warning" onClick={() => alert('You clicked the Button')}>
    Click me
  </Button>
  <Button type="danger" onClick={() => alert('You clicked the Button')}>
    Click me
  </Button>
  <Button type="text" onClick={() => alert('You clicked the Button')}>
    Click me
  </Button>
  <Button type="link" onClick={() => alert('You clicked the Button')}>
    Click me
  </Button>
</Playground>

### Link 按钮

<Playground>
  <Button href="https://github.com/billjs/magot" target="_blank">
    Link me
  </Button>
  <Button type="primary" href="https://github.com/billjs/magot" target="_blank">
    Link me
  </Button>
  <Button type="success" href="https://github.com/billjs/magot" target="_blank">
    Link me
  </Button>
  <Button type="warning" href="https://github.com/billjs/magot" target="_blank">
    Link me
  </Button>
  <Button type="danger" href="https://github.com/billjs/magot" target="_blank">
    Link me
  </Button>
  <Button type="text" href="https://github.com/billjs/magot" target="_blank">
    Link me
  </Button>
  <Button type="link" href="https://github.com/billjs/magot" target="_blank">
    Link me
  </Button>
</Playground>

### 自定义样式按钮

<Playground>
  <Button
    style={{
      color: '#52c41a',
      backgroundColor: '#343a40',
      borderColor: '#343a40',
    }}>
    My Button
  </Button>
  <Button
    style={{
      color: '#fff',
      backgroundColor: '#f5222d',
      borderColor: '#f5222d',
      borderRadius: 0,
    }}>
    My Button
  </Button>
</Playground>

### 按钮组合

文档详见：[按钮组合](/components/ButtonGroup)。

<Playground>
  <Button.Group>
    <Button>L</Button>
    <Button>M</Button>
    <Button>R</Button>
  </Button.Group>
  <Button.Group>
    <Button type="primary" icon="arrow-left">
      Go Back
    </Button>
    <Button type="primary" icon="arrow-right" iconPosition="right">
      Go Forward
    </Button>
  </Button.Group>
  <Button.Group>
    <Button type="primary" icon="download" square={true} />
    <Button type="primary" icon="upload" square={true} />
  </Button.Group>
  <Button.Group>
    <Button type="success" icon="smile" square={true} />
    <Button type="success" icon="view" square={true} />
  </Button.Group>
  <Button.Group>
    <Button type="warning" icon="alipay" square={true} />
    <Button type="warning" icon="wxpay" square={true} />
  </Button.Group>
  <Button.Group>
    <Button type="danger" icon="delete" square={true} />
    <Button type="danger" icon="close" square={true} />
  </Button.Group>
</Playground>

## Props

<PropsTable of={Button} />
