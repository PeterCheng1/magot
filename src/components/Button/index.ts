export { default, ButtonProps, ButtonType } from './Button';
