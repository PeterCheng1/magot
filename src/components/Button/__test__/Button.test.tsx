import React from 'react';
import { render, shallow } from 'enzyme';
import Button from '../Button';
import {
  injectStyleTestSuites,
  injectClickEventTestSuite,
} from '../../../../test/shared.test';

describe('Button', () => {
  it('should renders dom correctly', () => {
    const wrapper = render(<Button />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should renders a `button` Button', () => {
    const btn = <Button>Button</Button>;
    const wrapper1 = shallow(btn);
    expect(wrapper1.name()).toEqual('NativeButton');

    const wrapper2 = render(btn);
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper2.attr('type')).toEqual('button');
  });

  it('should renders a `link` Button', () => {
    const link = (
      <Button href="https://github.com/billjs/magot" target="_blank">
        Link
      </Button>
    );
    const wrapper1 = shallow(link);
    expect(wrapper1.name()).toEqual('LinkButton');

    const wrapper2 = render(link);
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper2.attr('href')).toEqual('https://github.com/billjs/magot');
    expect(wrapper2.attr('target')).toEqual('_blank');
  });

  it('should renders children when passed in', () => {
    let wrapper = render(<Button>This is a Button</Button>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toEqual('This is a Button');

    wrapper = render(
      <Button>
        <span>Button</span>
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('span').text()).toEqual('Button');

    wrapper = render(<Button href="#/home">Link</Button>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toEqual('Link');

    wrapper = render(
      <Button href="#/home">
        <span>Home</span>
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('span').text()).toEqual('Home');
  });

  it('should renders `type` prop when passed in', () => {
    let wrapper1 = render(<Button type="primary">Primary</Button>);
    let wrapper2 = render(
      <Button href="#" type="primary">
        Primary
      </Button>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.hasClass('mgt-button-primary')).toBe(true);
    expect(wrapper2.hasClass('mgt-button-primary')).toBe(true);

    wrapper1 = render(<Button type="success">Success</Button>);
    wrapper2 = render(
      <Button href="#" type="success">
        Success
      </Button>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.hasClass('mgt-button-success')).toBe(true);
    expect(wrapper2.hasClass('mgt-button-success')).toBe(true);

    wrapper1 = render(<Button type="warning">Warning</Button>);
    wrapper2 = render(
      <Button href="#" type="warning">
        Warning
      </Button>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.hasClass('mgt-button-warning')).toBe(true);
    expect(wrapper2.hasClass('mgt-button-warning')).toBe(true);

    wrapper1 = render(<Button type="danger">Danger</Button>);
    wrapper2 = render(
      <Button href="#" type="danger">
        Danger
      </Button>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.hasClass('mgt-button-danger')).toBe(true);
    expect(wrapper2.hasClass('mgt-button-danger')).toBe(true);

    wrapper1 = render(<Button type="text">Text</Button>);
    wrapper2 = render(
      <Button href="#" type="text">
        Text
      </Button>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.hasClass('mgt-button-text')).toBe(true);
    expect(wrapper2.hasClass('mgt-button-text')).toBe(true);

    wrapper1 = render(<Button type="link">Link</Button>);
    wrapper2 = render(
      <Button href="#" type="link">
        Link
      </Button>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.hasClass('mgt-button-link')).toBe(true);
    expect(wrapper2.hasClass('mgt-button-link')).toBe(true);
  });

  it('should renders `size` prop when passed in', () => {
    let wrapper1 = render(<Button size="small">Small</Button>);
    let wrapper2 = render(
      <Button href="#" size="small">
        Small
      </Button>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.hasClass('mgt-button-sm')).toBe(true);
    expect(wrapper2.hasClass('mgt-button-sm')).toBe(true);

    wrapper1 = render(<Button size="large">Large</Button>);
    wrapper2 = render(
      <Button href="#" size="large">
        Large
      </Button>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.hasClass('mgt-button-lg')).toBe(true);
    expect(wrapper2.hasClass('mgt-button-lg')).toBe(true);
  });

  it('should renders `disabled` prop when passed in', () => {
    const wrapper1 = render(<Button disabled={true}>Disabled</Button>);
    const wrapper2 = render(
      <Button href="#" disabled={true}>
        Disabled
      </Button>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.hasClass('mgt-button-disabled')).toBe(true);
    expect(wrapper2.hasClass('mgt-button-disabled')).toBe(true);
    expect(wrapper1.attr('disabled')).toEqual('disabled');
  });

  it('should renders `block` prop when passed in', () => {
    const wrapper1 = render(<Button block={true}>Block</Button>);
    const wrapper2 = render(
      <Button href="#" block={true}>
        Block
      </Button>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.hasClass('mgt-button-block')).toBe(true);
    expect(wrapper2.hasClass('mgt-button-block')).toBe(true);
  });

  it('should renders `circular` prop when passed in', () => {
    const wrapper1 = render(<Button circular={true} />);
    const wrapper2 = render(<Button href="#" circular={true} />);
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.hasClass('mgt-button-circular')).toBe(true);
    expect(wrapper2.hasClass('mgt-button-circular')).toBe(true);
  });

  it('should renders `square` prop when passed in', () => {
    const wrapper1 = render(<Button square={true} />);
    const wrapper2 = render(<Button href="#" square={true} />);
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.hasClass('mgt-button-square')).toBe(true);
    expect(wrapper2.hasClass('mgt-button-square')).toBe(true);
  });

  it('should renders with icon', () => {
    let wrapper1 = render(<Button icon="download">Icon Button</Button>);
    let wrapper2 = render(
      <Button href="#" icon="download">
        Icon Button
      </Button>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.find('.mgt-icon')).toHaveLength(1);
    expect(wrapper2.find('.mgt-icon')).toHaveLength(1);

    wrapper1 = render(
      <Button icon="download" iconPosition="right">
        Icon Button
      </Button>
    );
    wrapper2 = render(
      <Button href="#" icon="download" iconPosition="right">
        Icon Button
      </Button>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.find('.mgt-icon').hasClass('right')).toBe(true);
    expect(wrapper2.find('.mgt-icon').hasClass('right')).toBe(true);

    wrapper1 = render(
      <Button icon="download" iconSize={20}>
        Icon Button
      </Button>
    );
    wrapper2 = render(
      <Button href="#" icon="download" iconSize={20}>
        Icon Button
      </Button>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
  });

  it('should renders `loading` prop when passed in', () => {
    const wrapper1 = render(<Button loading={true} />);
    const wrapper2 = render(<Button href="#" loading={true} />);
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper1.find('.mgt-icon-loading.spin')).toHaveLength(1);
    expect(wrapper2.find('.mgt-icon-loading.spin')).toHaveLength(1);
  });

  it('should renders `htmlType` prop when passed in', () => {
    const wrapper = render(<Button htmlType="submit">Button</Button>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.attr('type')).toEqual('submit');
  });

  injectStyleTestSuites(Button);
  injectClickEventTestSuite(Button);
});
