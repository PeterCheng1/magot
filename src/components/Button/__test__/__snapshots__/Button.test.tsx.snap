// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should renders \`block\` prop when passed in 1`] = `
<button
  class="mgt-button mgt-button-block"
  type="button"
>
  Block
</button>
`;

exports[`Button should renders \`block\` prop when passed in 2`] = `
<a
  class="mgt-button mgt-button-block"
  href="#"
>
  Block
</a>
`;

exports[`Button should renders \`circular\` prop when passed in 1`] = `
<button
  class="mgt-button mgt-button-circular"
  type="button"
/>
`;

exports[`Button should renders \`circular\` prop when passed in 2`] = `
<a
  class="mgt-button mgt-button-circular"
  href="#"
/>
`;

exports[`Button should renders \`disabled\` prop when passed in 1`] = `
<button
  class="mgt-button mgt-button-disabled"
  disabled=""
  type="button"
>
  Disabled
</button>
`;

exports[`Button should renders \`disabled\` prop when passed in 2`] = `
<a
  class="mgt-button mgt-button-disabled"
  disabled=""
  href="#"
>
  Disabled
</a>
`;

exports[`Button should renders \`htmlType\` prop when passed in 1`] = `
<button
  class="mgt-button"
  type="submit"
>
  Button
</button>
`;

exports[`Button should renders \`loading\` prop when passed in 1`] = `
<button
  class="mgt-button"
  type="button"
>
  <i
    class="mgt-icon mgt-icon-loading spin"
  />
</button>
`;

exports[`Button should renders \`loading\` prop when passed in 2`] = `
<a
  class="mgt-button"
  href="#"
>
  <i
    class="mgt-icon mgt-icon-loading spin"
  />
</a>
`;

exports[`Button should renders \`size\` prop when passed in 1`] = `
<button
  class="mgt-button mgt-button-sm"
  type="button"
>
  Small
</button>
`;

exports[`Button should renders \`size\` prop when passed in 2`] = `
<a
  class="mgt-button mgt-button-sm"
  href="#"
>
  Small
</a>
`;

exports[`Button should renders \`size\` prop when passed in 3`] = `
<button
  class="mgt-button mgt-button-lg"
  type="button"
>
  Large
</button>
`;

exports[`Button should renders \`size\` prop when passed in 4`] = `
<a
  class="mgt-button mgt-button-lg"
  href="#"
>
  Large
</a>
`;

exports[`Button should renders \`square\` prop when passed in 1`] = `
<button
  class="mgt-button mgt-button-square"
  type="button"
/>
`;

exports[`Button should renders \`square\` prop when passed in 2`] = `
<a
  class="mgt-button mgt-button-square"
  href="#"
/>
`;

exports[`Button should renders \`type\` prop when passed in 1`] = `
<button
  class="mgt-button mgt-button-primary"
  type="button"
>
  Primary
</button>
`;

exports[`Button should renders \`type\` prop when passed in 2`] = `
<a
  class="mgt-button mgt-button-primary"
  href="#"
>
  Primary
</a>
`;

exports[`Button should renders \`type\` prop when passed in 3`] = `
<button
  class="mgt-button mgt-button-success"
  type="button"
>
  Success
</button>
`;

exports[`Button should renders \`type\` prop when passed in 4`] = `
<a
  class="mgt-button mgt-button-success"
  href="#"
>
  Success
</a>
`;

exports[`Button should renders \`type\` prop when passed in 5`] = `
<button
  class="mgt-button mgt-button-warning"
  type="button"
>
  Warning
</button>
`;

exports[`Button should renders \`type\` prop when passed in 6`] = `
<a
  class="mgt-button mgt-button-warning"
  href="#"
>
  Warning
</a>
`;

exports[`Button should renders \`type\` prop when passed in 7`] = `
<button
  class="mgt-button mgt-button-danger"
  type="button"
>
  Danger
</button>
`;

exports[`Button should renders \`type\` prop when passed in 8`] = `
<a
  class="mgt-button mgt-button-danger"
  href="#"
>
  Danger
</a>
`;

exports[`Button should renders \`type\` prop when passed in 9`] = `
<button
  class="mgt-button mgt-button-text"
  type="button"
>
  Text
</button>
`;

exports[`Button should renders \`type\` prop when passed in 10`] = `
<a
  class="mgt-button mgt-button-text"
  href="#"
>
  Text
</a>
`;

exports[`Button should renders \`type\` prop when passed in 11`] = `
<button
  class="mgt-button mgt-button-link"
  type="button"
>
  Link
</button>
`;

exports[`Button should renders \`type\` prop when passed in 12`] = `
<a
  class="mgt-button mgt-button-link"
  href="#"
>
  Link
</a>
`;

exports[`Button should renders a \`button\` Button 1`] = `
<button
  class="mgt-button"
  type="button"
>
  Button
</button>
`;

exports[`Button should renders a \`link\` Button 1`] = `
<a
  class="mgt-button"
  href="https://github.com/billjs/magot"
  target="_blank"
>
  Link
</a>
`;

exports[`Button should renders children when passed in 1`] = `
<button
  class="mgt-button"
  type="button"
>
  This is a Button
</button>
`;

exports[`Button should renders children when passed in 2`] = `
<button
  class="mgt-button"
  type="button"
>
  <span>
    Button
  </span>
</button>
`;

exports[`Button should renders children when passed in 3`] = `
<a
  class="mgt-button"
  href="#/home"
>
  Link
</a>
`;

exports[`Button should renders children when passed in 4`] = `
<a
  class="mgt-button"
  href="#/home"
>
  <span>
    Home
  </span>
</a>
`;

exports[`Button should renders custom className 1`] = `
<button
  class="mgt-button jest-test-class"
  type="button"
/>
`;

exports[`Button should renders custom styles 1`] = `
<button
  class="mgt-button"
  style="color:#333;background-color:#f0f0f0"
  type="button"
/>
`;

exports[`Button should renders dom correctly 1`] = `
<button
  class="mgt-button"
  type="button"
/>
`;

exports[`Button should renders with icon 1`] = `
<button
  class="mgt-button"
  type="button"
>
  <i
    class="mgt-icon mgt-icon-download left"
  />
  Icon Button
</button>
`;

exports[`Button should renders with icon 2`] = `
<a
  class="mgt-button"
  href="#"
>
  <i
    class="mgt-icon mgt-icon-download left"
  />
  Icon Button
</a>
`;

exports[`Button should renders with icon 3`] = `
<button
  class="mgt-button"
  type="button"
>
  Icon Button
  <i
    class="mgt-icon mgt-icon-download right"
  />
</button>
`;

exports[`Button should renders with icon 4`] = `
<a
  class="mgt-button"
  href="#"
>
  Icon Button
  <i
    class="mgt-icon mgt-icon-download right"
  />
</a>
`;

exports[`Button should renders with icon 5`] = `
<button
  class="mgt-button"
  type="button"
>
  <i
    class="mgt-icon mgt-icon-download left"
    style="font-size:20px"
  />
  Icon Button
</button>
`;

exports[`Button should renders with icon 6`] = `
<a
  class="mgt-button"
  href="#"
>
  <i
    class="mgt-icon mgt-icon-download left"
    style="font-size:20px"
  />
  Icon Button
</a>
`;
